cypress-io/cypress
# test: tweak simulated run mode and some other tests for flake

- Closes #23306 and does some further things.

### Additional details


#24846 changed the way `run` mode is determined, which introduced some flake in the `cypress-in-cypress-run-mode` tests which simulated `run` mode by modifying `window.__CYPRESS_MODE__`. 

The flake wasn't noticed right away, since the tests usually passed with retries, in a pretty accidental way.

This PR cleans up those tests and a few other instances of flake that were noticed along the way, and changes the way `run` mode is simulated in the cy-in-cy test. The `isRunMode` utility function added in #24846 can now return true if a special `CY_IN_CY_SIMULATE_RUN_MODE` parameter is added to the URL. This removed the race condition on startup between Cypress overriding the global variable and the Vue app initializing that I suspect was at the root of the flake.

This PR does a few other things as well, let's explain them:

- allows `CloudViewerAndProject` and `LoginConnectModals` to render always in open mode - since the navbar in open mode now shows some cloud data, and avoiding them in the spec-runner was mostly about avoiding graphql in run mode.
- modifies a few places in the UI that still checked `window.__CYPRESS_MODE__ !== 'run'` to use the `utils/isRunMode` function - this makes things more consistent and makes the cy-in-cy run mode tests behave better
- deletes `HideDuringScreenshotOrRunMode` component - this was hanging around from the time before `SpecRunnerOpenMode` and `SpecRunnerRunMode` were split up. It was still used in `SpecRunnerOpenMode` even though that is never rendered in `run` mode. This had confusing side effects in the simulated run mode.

### Steps to test


Run the tests in `packages/app/cypress/e2e/cypress-in-cypress-run-mode.cy.ts` locally and make sure they they pass and look correct. Verify we see no flake in CI.

Similarly, check `packages/app/src/runner/selector-playground/SelectorPlayground.cy.tsx` and see that it passes locally too.

### How has the user experience changed?


### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
